aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -69,35 +76,64 @@ public String getPluginKey() {
     public void configure(
             @NonNull JSONObject pluginConfiguration,
             @NonNull Context context
-    ) throws AmplifyException {
+    ) throws AuthException {
         final CountDownLatch latch = new CountDownLatch(1);
+        final AsyncResult<Exception> asyncException = new AsyncResult<>();
 
         AWSMobileClient.getInstance().initialize(
             context,
             new AWSConfiguration(pluginConfiguration),
             new Callback<UserStateDetails>() {
                 @Override
                 public void onResult(UserStateDetails result) {
-                    latch.countDown();
+                    if (result.getUserState().equals(UserState.SIGNED_IN)) {
+                        fetchAndSetUserId(() -> latch.countDown());
+                    } else {
+                        userId = null;
+                        latch.countDown();
+                    }
+
+                    AWSMobileClient.getInstance().addUserStateListener(userStateDetails -> {
+                        switch (userStateDetails.getUserState()) {
+                            case SIGNED_IN:
+                                fetchAndSetUserId(() -> { /* No response needed */ });
+                                break;
+                            default:
+                                userId = null;
+                        }
+                    });
                 }
 
                 @Override
                 public void onError(Exception error) {
-                    throw new IllegalStateException("Failed to instantiate AWSMobileClient", error);
+                    asyncException.set(error);
+                    latch.countDown();
                 }
             }
         );
 
         try {
```

Comment:
The three exceptions being created from within this try are all very similar. Think of some ways you can create one error, and throw it if any of the scenarios arrise.

Reply:
Also need clarification on this so I'll ask about it on our call